### PR TITLE
Added bundle scripts, smaples, ReadMe.md and ChangeLog.md for Browserify

### PR DESCRIPTION
**Pull request by XiaoningLiu**
Updated or added following items:
- Browserify bundle scripts
- Samples for JS Client Library
- README.md
- ChangeLog.md
- NodeJS doc

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@XiaoningLiu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
 * Master [![Build Status](https://travis-ci.org/Azure/azure-storage-node.svg?branch=master)](https://travis-ci.org/Azure/azure-storage-node/branches) [![Coverage Status](https://coveralls.io/repos/Azure/azure-storage-node/badge.svg?branch=master&service=github)](https://coveralls.io/github/Azure/azure-storage-node?branch=master)
 * Dev [![Build Status](https://travis-ci.org/Azure/azure-storage-node.svg?branch=dev)](https://travis-ci.org/Azure/azure-storage-node/branches) [![Coverage Status](https://coveralls.io/repos/Azure/azure-storage-node/badge.svg?branch=dev&service=github)](https://coveralls.io/github/Azure/azure-storage-node?branch=dev)
 
-This project provides a Node.js package that makes it easy to consume and manage Microsoft Azure Storage Services.
+This project provides a Node.js package and a browser compatible [JavaScript Client Library](#azure-storage-javascript-client-library-for-browsers) that make it easy to consume and manage Microsoft Azure Storage Services.
```
Comment (hasonmsft) [inline README.md]: makes

File: README.md [hunk 2]
```diff
@@ -452,6 +452,22 @@ set HTTP_PROXY=http://127.0.0.1:8888
 
 On Linux, please use `export` other than `set` to set the variables.
 
+# Azure Storage JavaScript Client Library for Browsers
+
+* JavaScript Client Library is currently in Preview. Questions and feedback are welcome in our Slack channel. [![Slack](https://azurestorageslack.azurewebsites.net/badge.svg)]( https://azurestorageslack.azurewebsites.net)
```
Comment (hasonmsft) [inline README.md]: [slack channel]( https://azurestorageslack.azurewebsites.net)

File: README.md [hunk 3]
```diff
@@ -452,6 +452,22 @@ set HTTP_PROXY=http://127.0.0.1:8888
...
+
+You can also choose to download the JavaScript Client Library provided by us, or generate the library by yourself. Please refer to the [README.md](browser/README.md) under `browser` folder for detailed usage guidelines.
+
+## Downloading Azure Storage JavaScript Client Library
+
+It's recommended to use the Azure Storage JavaScript Client Library provided by us. Please [download the latest library](https://aka.ms/downloadazurestoragejs) from Azure CDN. 
+
+## Generating Azure Storage JavaScript Client Library
+
+We also provide browserify bundle scripts which generate Azure Storage JavaScript Client Library. The bundle script reduces the size of the Storage Client Library by splitting into smaller files, one per storage service. The [README.md](browser/README.md) under `browser` folder describes detailed generating guidelines.
```
Comment (hasonmsft) [inline README.md]: one per storage service and a common shared file --- sth. like this.

Comment (hasonmsft) [inline README.md]: The [README.md](browser/README.md) under `browser` folder describes detailed generating guidelines. 
==>
For more detailed information, refer to [README.md](browser/README.md) under `browser` folder.

File: browser/ChangeLog.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Note: This is the change log file for Azure Storage JavaScript Client Library.
+
+2017.02 Version 2.0.0-prerelease.1
```
Comment (hasonmsft) [inline browser/ChangeLog.md]: * Generated browser compatible JavaScript files based on Microsoft Azure Storage SDK for Node.js 2.0.0

Comment (hasonmsft) [inline browser/ChangeLog.md]: 2017.03

File: browser/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+* Join the community discussion on Slack! [![Slack](https://azurestorageslack.azurewebsites.net/badge.svg)]( https://azurestorageslack.azurewebsites.net) 
+
+There are 5 generated JavaScript files for Azure Storage JavaScript Client Library:
+- `azure-storage.common.js` contains the common part for other 4 JavaScript files.
+- `azure-storage.table.js` contains the Azure Storage table service operation logic, which depends on azure-storage.common.js
+- `azure-storage.blob.js` contains the Azure Storage blob service operation logic, which depends on azure-storage.common.js
+- `azure-storage.queue.js` contains the Azure Storage queue service operation logic, which depends on azure-storage.common.js
+- `azure-storage.file.js` contains the Azure Storage file service operation logic, which depends on azure-storage.common.js
+
+We also provide samples to guide you quickly start with the Azure Storage JavaScript Client Library. Under your downloaded JavaScript Client Library zip file or [azure-storage-node/browser/samples](samples), you will find 4 HTML samples:
```
Comment (hasonmsft) [inline browser/README.md]: In the [JavaScript Client Library zip file](link...)

File: browser/README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+- `sample-table.html` demostrates how to operate with Azure Storage table service in the browser
+- `sample-blob.html` demostrates how to operate with Azure Storage blob service in the browser
+- `sample-queue.html` demostrates how to operate with Azure Storage queue service in the browser
+- `sample-file.html` demostrates how to operate with Azure Storage file service in the browser
+
+After the JavaScript Client Library generation, you could open directly and try the samples with browsers such as Chrome/Edge/Firefox etc.
```
Comment (hasonmsft) [inline browser/README.md]: After generating the JavaScript Client Library, you can try the samples in browsers such as Chrome/Edge/Firefox directly.

File: browser/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+Theses "local file" related APIs are not recommended in the browser because of the browsers sandbox limitation. The "local file" here is not actually the local file on your disk. Data of the "local file" is kept in the browser memory. It is wrappered with a browserify module [browserify-fs](https://www.npmjs.com/package/browserify-fs), which encapsulates file system APIs based on your browser's IndexedDB.
+
+The "local file" related APIs are kept in the JavaScript Client Library, because we want to keep the compatibility for the browserify of some native Node.js applications. Some Node.js applications depends on Azure Storage Node.js Client Library and the fs module to operate with local files on the disk. When browserifing these applications, browserify-fs helps the applications adapt to the browser environment.
+
+For blob and file uploading or downloading with Azure Storage JavaScript Client Library in browser, please refer to our samples.
+
+### Compatibility
+
+Compatibility with mobile browsers have not been fully validated, please open issues when you get errors. Current validated browsers are as follows:
```
Comment (hasonmsft) [inline browser/README.md]: as below

File: browser/README.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+| Chrome     | Firefox  | Internet Explorer  | Microsoft Edge  |
+|------------|----------|--------------------|-----------------|
+| v55        | v50      | v11                | v38             |
+| v56        | v51      |                    | v39             |
+
+## Generating a Custom Azure Storage JavaScript Client Library
+
+If you wish to customize the library and generate the Azure Storage JavaScript Client Library, you can follow the following steps.
+
+We provide browserify bundle scripts which generate Azure Storage JavaScript Client Library. The bundle script reduces the size of the Storage Client Library by splitting into smaller files, one per storage service.
```
Comment (hasonmsft) [inline browser/README.md]: same comment for another place.

File: README.md (hunk 1 above)
Comment (XiaoningLiu) [inline README.md]: a Node.js package **and** a ... library that make

**Comment by XiaoningLiu** [comment]
Updated ChangeLog.md according to comments.

**Comment by azurecla** [comment]
Hi __@XiaoningLiu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Pull request merged**